system/ee/ExpressionEngine/Addons/forum/mod.forum_core.php

```diff
@@ -5931,7 +5931,8 @@ public function change_status()
         $viewpath = ($query->row('announcement') == 'n') ? 'viewthread/' : 'viewannounce/';
         $viewpath .= ($query->row('announcement') == 'n') ? $_GET['topic_id'] : $_GET['topic_id'] . '_' . $query->row('forum_id') ;
 
-        if (!array_key_exists('csrf_token',$_GET) ||  ! $this->_mod_permission('can_change_status', $query->row('forum_id')) || $_GET['csrf_token'] != CSRF_TOKEN) {
+        $token = ee('Request')->get('csrf_token');
+        if ( (! bool_config_item('disable_csrf_protection') &&  !array_key_exists('csrf_token',$_GET)  )  ||  ! $this->_mod_permission('can_change_status', $query->row('forum_id')) ||  (! bool_config_item('disable_csrf_protection') && $token != CSRF_TOKEN)) {
```

Comment:
I believe this does what it needs to do, however I see some ways how it can be improved:
- `disable_csrf_protection` can be checked only once
- `!array_key_exists('csrf_token',$_GET)` is like unnecessary - `ee('Request')->get('csrf_token')` would return empty value if the GET variable is not set

Reply:
Thanks, good catch. I have made the changes requested to make the logic more simplified.